ValheimPlus/RPC/VPlusConfigSync.cs

```diff
@@ -18,15 +21,43 @@ public static void RPC_VPlusConfigSync(long sender, ZPackage configPkg)
                 string[] rawConfigData = File.ReadAllLines(ConfigurationExtra.ConfigIniPath);
                 List<string> cleanConfigData = new List<string>();
 
-                for (int i = 0; i < rawConfigData.Length; i++)
-                {
-                    if (rawConfigData[i].Trim().StartsWith(";") ||
-                        rawConfigData[i].Trim().StartsWith("#")) continue; //Skip comments
+                FileIniDataParser parser = new FileIniDataParser();
+                IniData configdata = parser.ReadFile(ConfigurationExtra.ConfigIniPath);
+
 
-                    if (rawConfigData[i].Trim().IsNullOrWhiteSpace()) continue; //Skip blank lines
+                foreach (var prop in typeof(Configuration).GetProperties())
+                {
+                    var keyName = prop.Name;
+                    var method = prop.PropertyType.GetMethod("HasNeedsServerSync", BindingFlags.Public | BindingFlags.FlattenHierarchy | BindingFlags.Instance);
```

Comment:
Did you confirm that the fields marked NeedsServerSync are accurate? We haven't used this since the old config enforcement system so it may not be 100% accurate. Hotkeys are the primary things we don't want to enforce on clients.

Reply:
Yes, I spot the issue by seeing my AdvancedBuildingModeConfiguration and AdvancedEditingModeConfiguration was reapplying. With that approach only the ServerSyncConfig are applied.

Reply:
Here log to show you the process after receiving configuration:

```
[Info   : Unity Log] 03/04/2021 02:22:20: -------------------- SENDING VPLUGCONFIGSYNC REQUEST

[Info   : Unity Log] 03/04/2021 02:22:20: client got keys 4

[Info   : Unity Log] 03/04/2021 02:22:20: client got location icons

[Info   : Unity Log] 03/04/2021 02:22:20: Icons:2

[Info   : Unity Log] 03/04/2021 02:22:20: Session auth respons callback

[Info   : Unity Log] 03/04/2021 02:22:32: Starting respawn

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Beehive]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: honeyProductionSpeed=1200

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: maximumHoneyPerBeehive=4

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Building]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: noInvalidPlacementRestriction=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: noWeatherDamage=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: maximumPlacementDistance=5

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Items]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: noTeleportPrevention=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: baseItemWeightReduction=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: itemStackMultiplier=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Fermenter]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: fermenterDuration=2400

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: fermenterItemsProduced=6

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Fireplace]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=true

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: onlyTorches=true

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Food]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: foodDurationMultiplier=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Furnace]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: maximumOre=10

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: maximumCoal=20

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: coalUsedPerProduct=2

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: productionSpeed=30

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Kiln]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: maximumWood=20

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: productionSpeed=30

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Player]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: baseMaximumWeight=300

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: baseMegingjordBuff=150

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: baseAutoPickUpRange=2

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: disableCameraShake=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: baseUnarmedDamage=10

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Stamina]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: dodgeStaminaUsage=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: encumberedStaminaDrain=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: jumpStaminaDrain=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: runStaminaDrain=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: sneakStaminaDrain=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: staminaRegen=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: staminaRegenDelay=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: swimStaminaDrain=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [StaminaUsage]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: axes=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: bows=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: clubs=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: knives=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: pickaxes=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: polearms=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: spears=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: swords=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: unarmed=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: hammer=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: hoe=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Time]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=true

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: totalDayTimeInSeconds=120

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: nightTimeSpeedMultiplier=100

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Ward]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: wardRange=20

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [StructuralIntegrity]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: disableStructuralIntegrity=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: wood=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: stone=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: iron=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: hardWood=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Experience]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: swords=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: knives=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: clubs=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: polearms=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: spears=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: blocking=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: axes=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: bows=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: fireMagic=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: frostMagic=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: unarmed=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: pickaxes=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: woodCutting=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: jump=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: sneak=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: run=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: swim=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Game]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled=false

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: gameDifficultyDamageScale=0.4

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: gameDifficultyHealthScale=0.4

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: extraPlayerCountNearby=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: setFixedPlayerCountTo=0

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: difficultyScaleRange=200

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: [Wagon]

[Info   : Unity Log] 03/04/2021 02:22:32: CONFIGDATA: enabled = false

[Info   : Unity Log] Loading config section Beehive
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Building
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Items
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Fermenter
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Fireplace
[Info   : Unity Log]  Loading key onlyTorches
[Info   : Unity Log] Loading config section Food
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Furnace
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Kiln
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Player
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Stamina
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section StaminaUsage
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Time
[Info   : Unity Log]  Loading key totalDayTimeInSeconds
[Info   : Unity Log]  Loading key nightTimeSpeedMultiplier
[Info   : Unity Log] Loading config section Ward
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section StructuralIntegrity
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Experience
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Game
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] Loading config section Wagon
[Info   : Unity Log]  Section not enabled
[Info   : Unity Log] 03/04/2021 02:22:32: Successfully synced VPlus configuration from server.
```